anni/src/subcommands/repo.rs

```diff
@@ -238,6 +246,73 @@ fn repo_get_vgmdb(options: &RepoGetVGMdb, manager: &RepositoryManager, get: &Rep
     Ok(())
 }
 
+#[derive(Parser, Debug, Clone)]
+pub struct RepoGetCue {
+    #[clap(short = 'p', long)]
+    path: PathBuf,
+    #[clap(short = 'k', long, about = ll ! {"repo-get-cue-keyword"})]
+    keyword: Option<String>,
+}
+
+#[handler(RepoGetCue)]
+fn repo_get_cue(
+    options: &RepoGetCue,
+    manager: &RepositoryManager,
+    get: &RepoGetAction,
+) -> anyhow::Result<()> {
+    let path = &options.path;
+
+    let s = fs::read_to_string(path)?;
+    let cue = Cuna::new(&s)?;
+    let mut album = match (cue.catalog(), options.keyword.as_ref()) {
+        // if catalog is found, fetch metadata from vgmdb
+        (Some(catalog), _) => search_album(&catalog.to_string()).await?,
+        // otherwise try to search with keyword
+        (None, Some(keyword)) => {
+            warn!(
+                "catalog is unavailable, trying to search vgmdb with keyword `{}`",
+                keyword
+            );
+            search_album(&keyword.to_string()).await?
+        }
+        // if none is available, try to search with `TITLE` filed in the cue file
+        (None, None) => match cue.title().first() {
+            Some(title) => {
+                warn!("catalog is unavailable, trying to search vgmdb with title `{}`, which may be inaccurate", title);
+                search_album(&title.to_string()).await?
+            }
+            None => ball!("repo-missing-metadata", path = path.display().to_string()),
+        },
+    };
+
+    if album.catalog().is_empty() {
+        ball!("repo-missing-metadata", path = path.display().to_string())
```

Comment:
`catalog.is_empty()` means the album is avaliable, but the catalog on VGMdb is `N/A`, which is common on some tokuten CDs. We can assign a random catalog to it, or allow user to specify a catalog with `catalog` argument(just like what VGMdb subcommand does).